### ISSUE TITLE
[Signalr] Guarantee callback for one receiver

### ISSUE THREAD

**Issue by MhAllan** [dotnet/aspnetcore#24366]
SignalR callbacks are fire and forget. If SignalR callback can return when the receiver is one we can make many use cases much simpler. Now, we can't tell if the user received the message even he looks online in our database. Think of one to one chat solution with SignalR as relay. Bob sends message to Alice, for that to work when Alice is offline, the SignalR service needs to do this
1- Save Bob message in DB
2- Callback Alice
3- Alice confirms
4- Remove Bob message from DB

That sounds too much and not intuitive! It should be the other way around: 
1- Send message to Alice
2- If failed put in DB

SignalR will be much greater if the callback delivery is guaranteed and also if the offline messages were possible

**Comment by halter73** [comment]
Hi @MhAllan. Thanks for filing an issue. If I'm understanding this correctly, you want to be able to wait on the server side for a client `.On(` callback to complete. I think this could be achieved by "returning values from client invocations" (#5280) once SignalR has that feature. Do you agree? If so, I plan to close this issue as a duplicate of #5280 and continue to use the older issue to track progress in this area.
